doc/man3/BIO_sendmmsg.pod

```diff
@@ -0,0 +1,160 @@
+of received datagrams to be determined. Unlike B<peer>, support for B<local>
+must be explicitly enabled on a B<BIO> before it can be used; see
+BIO_dgram_set_local_addr_enable(). If B<local> is non-NULL in a B<BIO_MSG> and
+support for B<local> has not been enabled, processing of that B<BIO_MSG> fails.
+
+B<peer> and B<local> should be set to NULL if they are not required. Support for
+B<local> may not be available on all platforms; on these platforms, these
+functions always fail if B<local> is non-NULL.
+
+On Windows, if B<local> is specified, the B<BIO_ADDR> it points to will be
+zeroed (address family set to zero) if a local address was not reported by the
+operating system. This occurs when a packet is received which was sent by the
+local system. This occurs regardless of whether the packet's destination address
+was the loopback address or the IP address of a local non-loopback interface.
+This is an OS-specific limitation.
+
+The B<stride> argument must be set to B<sizeof(BIO_MSG)>. This argument
+facilitates backwards compatibility if flags are added to B<BIO_MSG>. Callers
+must zero-initialize B<BIO_MSG>.
+
+B<num_msg> should be sent to the maximum number of messages to send or receive,
+which is also the length of the array pointed to by B<msg>.
+
+Unlike most BIO functions, these functions explicitly support multi-threaded
+use. Multiple concurrent writers and multiple concurrent readers are permitted
+in any combination. As such, these functions do not clear, set, or otherwise
+modify BIO retry flags. The return value must be used to determine whether an
+operation should be retried; see below.
+
+BIO_dgram_set_local_addr_enable() and BIO_dgram_get_local_addr_enable() control
+whether local address support is enabled. To enable local address support, call
+BIO_dgram_set_local_addr_enable() with an argument of 1. The call will fail if
+local address support is not available for the platform.
+BIO_dgram_get_local_addr_enable() retrieves the value set by
+BIO_dgram_set_local_addr_enable().
+
+BIO_dgram_get_local_addr_cap() determines if the B<BIO> is capable of supporting
+local addresses.
+
+BIO_IS_ERRNO() and BIO_UNPACK_ERRNO() are used with the return values of
+BIO_sendmmsg() and BIO_recvmmsg(); see below.
+
+=head1 RETURN VALUES
+
+BIO_sendmmsg() and BIO_recvmmsg() the number of messages sent or received. Where
+a positive value n is returned, all entries in the B<BIO_MSG> array from 0
+through n-1 inclusive have their B<data_len> and B<flags> fields updated with
+the results of the operation on that message. If the call was to recvmmsg() and
+the B<peer> or B<local> fields of that message are non-NULL, the B<BIO_ADDR>
+structures they point to are written with the relevant address.
+
+If an error occurs before any message has been sent or received, a negative
+value is returned. -2 is returned if the BIO_sendmmsg() or BIO_recvmmsg() method
+is not supported by the BIO; -3 is returned if a local has been specified or
```

Comment:
I do not think the -3 return is a good idea, I'd just use -1. You have the `BIO_dgram_get_local_addr_cap()` to properly check.